src/Components/3DV/SceneView.tsx

```diff
@@ -1653,6 +1660,205 @@ function SceneView(props: ISceneViewProps, ref) {
         };
     }, [outlinedMeshitems, meshMap.current]);
 
+    // Handle gizmoElementItem
+    useEffect(() => {
+        debugLog(
+            'debug',
+            'adding gizmo to element based on gizmoElementItem prop' +
+                (scene ? ' with scene' : ' no scene')
+        );
+
+        if (scene && !isLoading) {
+            if (debugLogging) {
+                console.time('adding gizmo to meshes');
+            }
+            try {
+                // create a gizmoManager if one does not already exist
+                if (!gizmoManagerRef.current) {
+                    gizmoManagerRef.current = new BABYLON.GizmoManager(
+                        scene,
+                        1,
+                        utilLayer.current
+                    );
+                }
+                const gizmoManager = gizmoManagerRef.current;
+
+                if (!gizmoElementItem) {
```

Comment:
The contents of this if block are a good candidate for the cleanup + removing the gizmo manager refs.

Reply:
still need this logic for switching in and out of transforms mode, but we discussed the clean up together - done